### PR TITLE
Initial support for Nodejs 8

### LINKED ISSUE #2077: Add NodeJS 8 initial support for Serverless OpenWhisk

### PR DESCRIPTION
**Pull request by csantanapr**
Closes #2077 

Tracking nodejs@8 release here https://github.com/nodejs/node/issues/10117

### CODE REVIEW

File: core/nodejs8Action/build.gradle [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    }
+}
+
+ext.dockerImageName = 'nodejs8action'
+apply from: '../../gradle/docker.gradle'
+distDocker.dependsOn 'copyProxy'
+distDocker.finalizedBy('rmProxy')
+
+task copyProxy(type: Copy) {
+    from '../nodejsActionBase'
```
Comment (markusthoemmes) [inline core/nodejs8Action/build.gradle]: Want to use the same method as outlined for the python runners?

Comment (csantanapr) [inline core/nodejs8Action/build.gradle]: Good feedback, I will change it.

I took the example from swift3 runner
https://github.com/openwhisk/openwhisk/blob/master/core/swift3Action/build.gradle

Comment (csantanapr) [inline core/nodejs8Action/build.gradle]: @markusthoemmes need some help here:
made changes to 
```
ext.dockerImageName = 'nodejs8action'
apply from: '../../gradle/docker.gradle'
distDocker.dependsOn 'copyFiles'
distDocker.finalizedBy 'rmFiles'

def runners = files(
    new File(project(':core:nodejsActionBase').projectDir)
)

task copyFiles(type: Copy) {
    from runners
    into '.'
}

task rmFiles(type: Delete) {
    delete runners.collect { it.getName() }
}
```
But I get errors running `distDocker`
```
* What went wrong:
A problem occurred evaluating project ':core:nodejs8Action'.
> Could not find matching constructor for: java.io.File(java.io.File)

```

Comment (csantanapr) [inline core/nodejs8Action/build.gradle]: nevermind figured it out
```
project(':core:nodejsActionBase').projectDir
```

**Comment by csantanapr** [comment]
The nodejs CTC has voted to move the release date back a few weeks in order to give more time for V8 5.8 among other things. The new target release date for 8.0.0 is May 30th. The new semver-major cutoff date will be May 9th

**Comment by jthomas** [comment]
It is ready.
https://nodejs.org/en/blog/release/v8.0.0/

**Comment by csantanapr** [comment]
Yep I know

**Comment by csantanapr** [comment]
adding nodejs8 will be done in the new repo

**Pull request closed without merging**